dynamicio/mixins.py

```diff
@@ -672,28 +676,64 @@ def _read_from_postgres(self) -> pd.DataFrame:
 
         connection_string = f"postgresql://{db_user}:{db_password}@{db_host}:{db_port}/{db_name}"
 
-        model = self.options.get("model")
         sql_query = self.options.get("sql_query")
 
         query = None
 
-        if model and sql_query:
-            raise ValueError("Only one of `model` and `sql_query` must be provided")
-        if model is None and sql_query is None:
-            raise ValueError("One of `model` and `sql_query` must be provided")
+        if self.options.get("model"):
+            raise ValueError("`model` must not be provided")
 
-        if model:
-            query = Query(self._get_table_columns(model))
+        if "schema" not in self.sources_config:
+            schema_dict = self.schema
```

Comment:
Nice, accounting for the case where a schema is not explicitly provided as a definition.